File: TaylorRekker/TaylorForSqrt.m
% Skript som plottar Taylor-polynoma, opp til 
% ein bestemt orden, for funksjonen
% f(x) = sqrt(x) omkring x=1.
% Den maksimale ordenen til funksjonen er input.

% Gir maksimal orden
Nmax = 25;

% Funksjonen
funk=@(x) sqrt(x);

% Vektor med x-verdiar
x=0:1e-2:3;

% Intervall på y-aksen
ymin = -0.1;
ymax = 3;

% Initerar polynomet og koeffisienten i potensrekka
P=0*x;
teljar = 1;
nemnar = 1;

% Startar plot
figure(1)
plot(x,funk(x),'k-','linewidth',2)
axis([min(x) max(x) ymin ymax])
grid on
pause

% Første to polynom
% n = 0
P = teljar/nemnar * (x-1).^0;
plot(x,funk(x),'k-','linewidth',2)
grid on
hold on
plot(x,P,'r--','linewidth',2)
hold off
axis([min(x) max(x) ymin ymax])
title(['n=',num2str(n)])
pause

% n = 1
nemnar = 2;
P = P + teljar/nemnar * (x-1).^1;
plot(x,funk(x),'k-','linewidth',2)
grid on
hold on
plot(x,P,'r--','linewidth',2)
hold off
axis([min(x) max(x) ymin ymax])
title(['n=',num2str(n)])
pause

% Loopar over resten
for n=2:Nmax
  teljar = teljar*(2*n-3);
  nemnar = nemnar*(2*n);
  P=P + (-1)^(n+1)*teljar/nemnar*(x-1).^n;
  plot(x,funk(x),'k-','linewidth',2)
  grid on
  hold on
  plot(x,P,'r--','linewidth',2)
  hold off
  axis([min(x) max(x) ymin ymax])
  title(['n=',num2str(n)])
  pause
end
hold off

File: TaylorRekker/TaylorForSqrt.py
"""Skript som plottar Taylor-polynoma, opp til 
 ein bestemt orden, for funksjonen
 f(x) = sqrt(x) omkring x=1.
 Den maksimale ordenen til funksjonen er input."""

# Bibliotek
import numpy as np
import math
import matplotlib.pyplot as plt

# Maksimal orden
Nmax = 15

# Funksjonen
def funk(x):
    return np.sqrt(x)

# Vektor med x-verdiar
x = np.linspace(0, 3, 200)

# Initierer polynomet
P = 0*x

# Initerar tejlar og nemnar for koeffisentane
teljar = 1
nemnar = 1

# Startar plot
fig = plt.figure(1)

# Går gjennom alle N-verdiane til og med Nmax
for n in range(Nmax+1):
    # Neste ledd i Taylor-polynomet
    an = 1/math.factorial(n)
    P = P + teljar/nemnar*(x-1)**n

    plt.clf()   # Tømmer figuren

    plt.plot(x, funk(x), 'k-', linewidth=2)     # Plottar sjølve funksjonen

    # Plottar Taylor-polynomet
    plt.plot(x, P, 'r--', linewidth=2)          # Plottar tilnærminga
    
    plt.ylim(-0.1, 2)                              # Grenser for y-aksen
    plt.grid(True)                              # Set på rutenett
    plt.title(f'n = {n}')                       # Tittel på figuren

    # Ventar på tastetrykk
    plt.show(block=False)
    plt.pause(0.1)
    plt.waitforbuttonpress()
    
    # Oppdaterar teljar og nemnar
    teljar = -teljar*(2*n-1)
    nemnar = nemnar*2*(n+1)